playground/backend/cmd/server/controller.go

```diff
@@ -16,19 +16,77 @@ package main
 
 import (
 	pb "beam.apache.org/playground/backend/internal/api"
+	"beam.apache.org/playground/backend/internal/cache"
+	"beam.apache.org/playground/backend/internal/environment"
+	"beam.apache.org/playground/backend/internal/errors"
+	"beam.apache.org/playground/backend/internal/executors"
+	"beam.apache.org/playground/backend/internal/fs_tool"
 	"context"
 	"github.com/google/uuid"
+	"log"
+	"os"
+	"time"
 )
 
 type playgroundController struct {
 	pb.UnimplementedPlaygroundServiceServer
 }
 
+var cacheService = cache.NewCache(os.Getenv("cache"))
+
 //RunCode is running code from requests using a particular SDK
 func (controller *playgroundController) RunCode(ctx context.Context, info *pb.RunCodeRequest) (*pb.RunCodeResponse, error) {
-	// TODO implement this method
-	pipelineInfo := pb.RunCodeResponse{PipelineUuid: uuid.NewString()}
+	defer func() {
+		log.Println("RunCode complete")
+	}()
+
+	pipelineId := uuid.New()
+	env := environment.NewEnvironment()
```

Comment:
Why did you create a new environment here? That's static at the moment when the server is starting. Just keep it as an argument in the controller constructor.

Reply:
Changed.